src/Enhavo/Bundle/DashboardBundle/Tests/Provider/Type/TotalDashboardProviderTypeTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+
+namespace Enhavo\Bundle\DashboardBundle\Tests\Provider\Type;
+
+
+use Doctrine\ORM\AbstractQuery;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\QueryBuilder;
+use Enhavo\Bundle\DashboardBundle\Provider\Type\DashboardProviderType;
+use Enhavo\Bundle\DashboardBundle\Provider\Type\TotalDashboardProviderType;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class TotalDashboardProviderTypeTest extends TestCase
+{
+    public function testGetData()
+    {
+        $instance = $this->createInstance();
+        $parent = $this->createParentMock();
+
+        $parent->method('getRepository')->willReturnCallback(
```

Comment:
Looks like spaghetti. Better define the lowest mock first and pass it to the next higher mock for better code understanding. The `TotalDashboardProviderType` look also like a good candidate for functional testing against a sqlite database